app/data_providers/dispatch.py

```diff
@@ -256,15 +259,25 @@ async def _get_topic_slate_promises(
 
 class NewTabDispatch:
 
-    def __init__(self, new_tab_slate_provider: NewTabSlateProvider):
+    def __init__(self, new_tab_slate_provider: NewTabSlateProvider, snowplow: SnowplowCorpusRecommendationsTracker):
         self.new_tab_slate_provider = new_tab_slate_provider
+        self.snowplow = snowplow
 
-    async def get_slate(self) -> CorpusSlateModel:
+    async def get_slate(self, api_client: ApiClient) -> CorpusSlateModel:
         """
         :return: the New Tab slate
         """
-        return await self.new_tab_slate_provider.get_slate()
+        corpus_slate = await self.new_tab_slate_provider.get_slate()
+
+        asyncio.create_task(
+            self.snowplow.track(event=CorpusRecommendationsSendEvent(
```

Comment:
I like how we don't block execution by sending the tracking to a separate task as a secondary functionality! NewTab recs are critical and we should minimize the impact of broken external systems.

Reply:
Totally agreed. Most of my work in this project will be to ensure we meet the desired level of reliability. That includes SLO documentation, monitoring & observability tooling, incident workbooks, gracefull fallbacks (that are tested), and supporting end-to-end QA.